pgxn/neon/libpagestore.c

```diff
@@ -64,6 +65,20 @@ bool	(*old_redo_read_buffer_filter) (XLogReaderState *record, uint8 block_id) =
 
 static bool pageserver_flush(void);
 
+static pqsigfunc	 prev_signal_handler;
+
+static void
+pageserver_sighup_handler(SIGNAL_ARGS)
+{
+	if (prev_signal_handler)
+	{
+        	prev_signal_handler(postgres_signal_arg);
+	}
+	neon_log(LOG, "Received SIGHUP, flushing existing pageserver. New pageserver connstring is %s", page_server_connstring);
+	pageserver_flush();
```

Comment:
Why it is necessary to flush buffered requests to PS?
We are in any case not going to receive responses, aren't we?

Reply:
Good point, maybe just disconnect is sufficient?